Fixed Samsung devices

grzesiek2010 opened the pull request:
Closes #1424

#### Why is this the best possible solution? Were any other approaches considered?
As I said in the issue description it was visible on Collect 1.9.0 and Collect 1.9.1 but now there are many more instances so I'm pretty sure it's related to https://github.com/opendatakit/collect/pull/1399/files#diff-71ee66035a9814fa3226aa0ab0ddfd23R196 and I wasn't reproduce the issue. That's why I added [this](https://github.com/opendatakit/collect/pull/1425/files#diff-71ee66035a9814fa3226aa0ab0ddfd23R199) so now it works as before (using default dialog theme) for Samsungs API 21 and 22.

But I was able to reproduce this issue using TalkBack option and french language as described [here](https://stackoverflow.com/a/34853067/5479029)  and that's why I added https://github.com/opendatakit/collect/pull/1425/files#diff-71ee66035a9814fa3226aa0ab0ddfd23R202

I doubt someone uses TalkBack with ODK Collect as it's really hard but maybe someone has tried and that's why we can see similar crashes on 1.9.0 and 1.9.1.

#### Are there any risks to merging this code? If so, what are they?
It looks safe.

grzesiek2010 committed:
Fixed Samsung devices

lognaturel commented:
Nice find on the StackOverflow post! [This comment](https://stackoverflow.com/questions/28618405/datepicker-crashes-on-my-device-when-clicked-with-personal-app#comment52338909_31855744) suggests the problem is _not_ only limited to when talkback is on. Were you able to try a device type you are seeing in Firebase logs? If going by this approach, I would prefer not to check if talkback is on. Your description suggests you have reverted behavior for all Samsungs API 21 and 22 but in fact it's only if talkback is on. All  this makes me wonder -- why did you choose not to use [this approach](https://stackoverflow.com/a/34853067/137744)?

grzesiek2010 commented:
> This comment suggests the problem is not only limited to when talkback 

Yes I know and as I said I think that we can see more instances of this because of https://github.com/opendatakit/collect/pull/1399/files#diff-71ee66035a9814fa3226aa0ab0ddfd23R196 previously there were fewer instances. That's why I decided to use a default theme value for Samsungs API 21 and 22 https://github.com/opendatakit/collect/pull/1425/files#diff-71ee66035a9814fa3226aa0ab0ddfd23R199 

The problem with talkback I fixed here https://github.com/opendatakit/collect/pull/1425/files#diff-71ee66035a9814fa3226aa0ab0ddfd23R202

> Were you able to try a device type you are seeing in Firebase logs?

I used Samsung with API 21 and I was able to reproduce only the problem with talkback.

> Your description suggests you have reverted behavior for all Samsungs API 21 and 22 but in fact it's only if talkback is on

No look here https://github.com/opendatakit/collect/pull/1425/files#diff-71ee66035a9814fa3226aa0ab0ddfd23R199 I reverted behavior for all Samsungs  when it comes to the calendar appearance and additionally I we have a Samsung with API 21 or 22 and talkback is on I force to show spinners instead of calendar to avoid the crash I'm able to reproduce.

> All this makes me wonder -- why did you choose not to use this approach?

Because the main problem in my opinion is not related to talkback (I think it's rare). Te problem is now we use android.R.style.Theme_Material_Light_Dialog instead of default value for those Samsungs. Since I'm not able to reproduce the issue (without enabling talkback) I thought it's more reliable to use the version we used before and not testing this solution. What do you think? Should we try?

lognaturel commented:
Thanks for the additional clarification, I see now that the material design theme is an override of an existing theme that you are avoiding for Samsung devices with Android 21/22.

My understanding from reading the StackOverflow thread is that the second approach also addresses the non-talkback case. But let's try this and see if it helps.
